--- PCL-1.4.0-Source.old/CMakeLists.txt	2012-01-01 20:39:43.000000000 -0500
+++ PCL-1.4.0-Source/CMakeLists.txt	2012-02-20 19:37:03.000000000 -0500
@@ -128,13 +128,17 @@
 # OpenMP
 include(${PCL_SOURCE_DIR}/cmake/pcl_openmp.cmake)
 # OpenNI
-find_package(OpenNI)
+if(WITH_OPENNI)
+  find_package(OpenNI)
+endif()
 # Qhull
 find_package(Qhull)
 # Find ROS
 include(${PCL_SOURCE_DIR}/cmake/pcl_find_ros.cmake)
 # Find VTK
-find_package(VTK)
+if(WITH_VTK)
+  find_package(VTK)
+endif()
 if(VTK_FOUND)
   set(VTK_FOUND TRUE)
   message(STATUS "VTK found (include: ${VTK_INCLUDE_DIRS}, lib: ${VTK_LIBRARY_DIRS})")
@@ -142,7 +146,9 @@
   set(HAVE_VTK ON)
 endif()
 #Find MPI
-find_package(MPI)
+if(WITH_MPI)
+  find_package(MPI)
+endif()
 if(MPI_CXX_FOUND)
   include_directories(${MPI_INCLUDE_PATH})
 endif(MPI_CXX_FOUND)
