From 4feb073dad67126f0ade035e94b668f11fbb4304 Mon Sep 17 00:00:00 2001
From: lordgeorg <lordgeorg@googlemail.com>
Date: Tue, 12 Mar 2019 22:44:28 +0100
Subject: [PATCH] Sets version to 9999

---
 CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6556346a..ba5b199f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -54,11 +54,11 @@ set(META_PROJECT_DESCRIPTION "Strict OpenGL objects wrapper.")
 set(META_AUTHOR_ORGANIZATION "CG Internals GmbH")
 set(META_AUTHOR_DOMAIN       "https://github.com/cginternals/globjects/")
 set(META_AUTHOR_MAINTAINER   "opensource@cginternals.com")
-set(META_VERSION_MAJOR       "2")
+set(META_VERSION_MAJOR       "9999")
 set(META_VERSION_MINOR       "0")
 set(META_VERSION_PATCH       "0")
 set(META_VERSION_REVISION    "${GIT_REV}")
-set(META_VERSION             "${META_VERSION_MAJOR}.${META_VERSION_MINOR}.${META_VERSION_PATCH}")
+set(META_VERSION             "9999")
 set(META_NAME_VERSION        "${META_PROJECT_NAME} v${META_VERSION} (${META_VERSION_REVISION})")
 set(META_CMAKE_INIT_SHA      "6a6f30b38a1cee31ccac3f091816f28e0f6bce4b")
 
-- 
2.21.0

