--- make.targets.orig	2007-10-30 20:14:07.000000000 -0400
+++ make.targets	2007-10-30 23:53:29.000000000 -0400
@@ -16,8 +16,8 @@
 	$(INSTALL) -m 644 $(HFILES) $(EXTRA_HFILES) $(includedir)/gandalf/$(PACKAGE)
 
 installsubh:: ../$(SUBPACKAGE).h
-	if [ ! -d $(includedir)/gandalf/$(PACKAGE)/$(SUBPACKAGE) ]; then mkdir $(includedir)/gandalf/$(PACKAGE)/$(SUBPACKAGE); fi
-	$(INSTALL) -m 644 $(HFILES) $(EXTRA_HFILES) $(includedir)/gandalf/$(PACKAGE)/$(SUBPACKAGE)
+	if [ ! -d $(DESTDIR)$(includedir)/gandalf/$(PACKAGE)/$(SUBPACKAGE) ]; then mkdir $(DESTDIR)$(includedir)/gandalf/$(PACKAGE)/$(SUBPACKAGE); fi
+	$(INSTALL) -m 644 $(HFILES) $(EXTRA_HFILES) $(DESTDIR)$(includedir)/gandalf/$(PACKAGE)/$(SUBPACKAGE)
 
 docsource:: $(CFILES) $(HFILES)
 	../preprocess ../.. $(CFILES) $(HFILES)
