--- a/Makefile	2020-08-10 02:30:51.000000000 -0700
+++ a/Makefile	2020-08-11 14:00:08.944195557 -0700
@@ -136,7 +136,7 @@
 # is just a symlink to your actual ABC working directory, as 'make mrproper'
 # will remove the 'abc' directory and you do not want to accidentally
 # delete your work on ABC..
-ABCREV = 341db25
-ABCPULL = 1
+ABCREV = default
+ABCPULL = 0
 ABCURL ?= https://github.com/YosysHQ/abc
 ABCMKARGS = CC="$(CXX)" CXX="$(CXX)" ABC_USE_LIBSTDCXX=1
