--- a/src/Mod/Draft/Resources/Draft.qrc	2020-11-07 15:46:55.718595834 +0100
+++ b/src/Mod/Draft/Resources/Draft.qrc	2020-11-07 16:50:32.249859844 +0100
@@ -88,7 +88,7 @@
         <file>icons/Snap_Perpendicular.svg</file>
         <file>icons/Snap_Special.svg</file>
         <file>icons/Snap_WorkingPlane.svg</file>
-        <file>patterns/brick01</file>
+        <file>patterns/brick01.svg</file>
         <file>patterns/concrete.svg</file>
         <file>patterns/cross.svg</file>
         <file>patterns/diagonal1.svg</file>
