Index: linux-4.0.0-gentoo/drivers/acpi/osl.c
===================================================================
--- linux-4.0.0-gentoo.orig/drivers/acpi/osl.c
+++ linux-4.0.0-gentoo/drivers/acpi/osl.c
@@ -151,7 +151,7 @@ static u32 acpi_osi_handler(acpi_string
 			osi_linux.cmdline ? " via cmdline" :
 			osi_linux.dmi ? " via DMI" : "");
 	}
-
+#if 0
 	if (!strcmp("Darwin", interface)) {
 		/*
 		 * Apple firmware will behave poorly if it receives positive
@@ -161,6 +161,7 @@ static u32 acpi_osi_handler(acpi_string
 		acpi_update_interfaces(ACPI_DISABLE_ALL_VENDOR_STRINGS);
 		supported = ACPI_UINT32_MAX;
 	}
+#endif
 
 	return supported;
 }
