--- Lib/distutils/ccompiler.py	2008-04-19 14:13:24.000000000 +0300
+++ Lib/distutils/ccompiler.py	2008-04-19 14:18:10.000000000 +0300
@@ -640,13 +640,16 @@
                 self._setup_compile(output_dir, macros, include_dirs, sources,
                                     depends, extra_postargs)
         cc_args = self._get_cc_args(pp_opts, debug, extra_preargs)
+        
+        lang = self.detect_language(sources)
 
         for obj in objects:
             try:
                 src, ext = build[obj]
             except KeyError:
                 continue
-            self._compile(obj, src, ext, cc_args, extra_postargs, pp_opts)
+            self._compile(obj, src, ext, cc_args, extra_postargs, pp_opts,
+                    lang=lang)
 
         # Return *all* object filenames, not just the ones we just built.
         return objects
--- Lib/distutils/unixccompiler.py	2008-04-19 14:13:32.000000000 +0300
+++ Lib/distutils/unixccompiler.py	2008-04-19 14:21:36.000000000 +0300
@@ -112,6 +112,7 @@
                    'compiler'     : ["cc"],
                    'compiler_so'  : ["cc"],
                    'compiler_cxx' : ["cc"],
+                   'compiler_cxx_so' : ["cc"],
                    'linker_so'    : ["cc", "-shared"],
                    'linker_exe'   : ["cc"],
                    'archiver'     : ["ar", "-cr"],
@@ -162,12 +163,18 @@
             except DistutilsExecError as msg:
                 raise CompileError(msg)
 
-    def _compile(self, obj, src, ext, cc_args, extra_postargs, pp_opts):
+    def _compile(self, obj, src, ext, cc_args, extra_postargs, pp_opts,
+            lang='c'):
         compiler_so = self.compiler_so
+        compiler_cxx_so = self.compiler_cxx_so
         if sys.platform == 'darwin':
             compiler_so = _darwin_compiler_fixup(compiler_so, cc_args + extra_postargs)
         try:
-            self.spawn(compiler_so + cc_args + [src, '-o', obj] +
+            if lang == 'c++':
+                cc = compiler_cxx_so + cc_args
+            else:
+                cc = compiler_so + cc_args
+            self.spawn(cc + cc_args + [src, '-o', obj] +
                        extra_postargs)
         except DistutilsExecError as msg:
             raise CompileError(msg)
--- Lib/distutils/sysconfig.py.orig	2008-03-19 08:28:24.000000000 +0200
+++ Lib/distutils/sysconfig.py	2008-04-19 18:29:24.000000000 +0300
@@ -162,9 +162,9 @@
     varies across Unices and is stored in Python's Makefile.
     """
     if compiler.compiler_type == "unix":
-        (cc, cxx, opt, cflags, ccshared, ldshared, so_ext) = \
-            get_config_vars('CC', 'CXX', 'OPT', 'CFLAGS',
-                            'CCSHARED', 'LDSHARED', 'SO')
+        (cc, cxx, opt, cflags, basecflags, ccshared, ldshared, so_ext, cxxflags) = (
+            get_config_vars('CC', 'CXX', 'OPT', 'CFLAGS', 'BASECFLAGS',
+                            'CCSHARED', 'LDSHARED', 'SO', 'CXXFLAGS'))
 
         if 'CC' in os.environ:
             cc = os.environ['CC']
@@ -179,19 +179,25 @@
         if 'LDFLAGS' in os.environ:
             ldshared = ldshared + ' ' + os.environ['LDFLAGS']
         if 'CFLAGS' in os.environ:
-            cflags = opt + ' ' + os.environ['CFLAGS']
+            cflags = basecflags + ' ' + os.environ['CFLAGS']
             ldshared = ldshared + ' ' + os.environ['CFLAGS']
         if 'CPPFLAGS' in os.environ:
             cpp = cpp + ' ' + os.environ['CPPFLAGS']
             cflags = cflags + ' ' + os.environ['CPPFLAGS']
             ldshared = ldshared + ' ' + os.environ['CPPFLAGS']
+        if 'CXXFLAGS' in os.environ:
+            cxxflags = opt + ' ' + os.environ['CXXFLAGS']
 
         cc_cmd = cc + ' ' + cflags
+        cxx_cmd = cxx
+        if cxxflags is not None:
+            cxx_cmd = cxx + ' ' + cxxflags
         compiler.set_executables(
             preprocessor=cpp,
             compiler=cc_cmd,
             compiler_so=cc_cmd + ' ' + ccshared,
             compiler_cxx=cxx,
+            compiler_cxx_so=cxx_cmd + ' ' + ccshared,
             linker_so=ldshared,
             linker_exe=cc)
 
@@ -533,6 +539,7 @@
                 for key in ('LDFLAGS', 'BASECFLAGS',
                         # a number of derived variables. These need to be
                         # patched up as well.
+                        'CXXFLAGS',
                         'CFLAGS', 'PY_CFLAGS', 'BLDSHARED'):
                     flags = _config_vars[key]
                     flags = re.sub('-arch\s+\w+\s', ' ', flags)
