--- a/net/socket.c	2009-08-13 19:13:00.000000000 -0400
+++ b/net/socket.c	2009-08-13 19:14:38.000000000 -0400
@@ -736,7 +736,7 @@ static ssize_t sock_sendpage(struct file
 	if (more)
 		flags |= MSG_MORE;
 
-	return sock->ops->sendpage(sock, page, offset, size, flags);
+	return kernel_sendpage(sock, page, offset, size, flags);
 }
 
 static ssize_t sock_splice_read(struct file *file, loff_t *ppos,
