diff -Nur mbot.orig/src/Makefile.in mbot/src/Makefile.in
--- mbot.orig/src/Makefile.in	2004-08-19 23:00:30.000000000 +0000
+++ mbot/src/Makefile.in	2007-07-22 00:07:09.845049911 +0000
@@ -20,11 +20,11 @@
 	$(CXX) $(CXXFLAGS) -I$(INCLUDE) $< -c
 
 mbot:	$(OBJS)
-	$(CXX) -s $(LDFLAGS) $(LIBS) $(OBJS) -o $(BIN)
+	$(CXX) -s $(LDFLAGS) $(OBJS) $(LIBS) -o $(BIN)
 
 static: $(OBJS)
-	$(CXX) -s $(LDFLAGS) -static $(LIBS) $(OBJS) -o $(BIN)
+	$(CXX) -s $(LDFLAGS) -static $(OBJS) $(LIBS) -o $(BIN)
 
 debug:  $(OBJS)
-	$(CXX) $(LDFLAGS) $(LIBS) $(OBJS) -o $(BIN)
+	$(CXX) $(LDFLAGS) $(OBJS) $(LIBS) -o $(BIN)
 
