From d21215cff394f8ad681b6c2f959d51474caa0765 Mon Sep 17 00:00:00 2001
From: "Yen-Chin, Lee" <coldnew.tw@gmail.com>
Date: Mon, 20 Jul 2020 00:26:27 +0800
Subject: [PATCH] fix-gentoo-clang-header

Signed-off-by: Yen-Chin, Lee <coldnew.tw@gmail.com>
---
 CMakeLists.txt | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index fd24771b..2c0c79b9 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -138,7 +138,8 @@ if (BUILD_CXX_LANGUAGE_PACKAGE)
 	set (CLANG_COMPILER_HEADER_SEARCH_PATH "")
 	foreach(LLVM_CONFIGURATION_TYPE ${LLVM_CONFIGURATION_TYPES})
 		if (CLANG_COMPILER_HEADER_SEARCH_PATH STREQUAL "")
-			set (CLANG_COMPILER_HEADER_SEARCH_PATH "${LLVM_BINARY_DIR}/${LLVM_CONFIGURATION_TYPE}/lib/clang/${LLVM_VERSION}/include")
+			#set (CLANG_COMPILER_HEADER_SEARCH_PATH "${LLVM_BINARY_DIR}/${LLVM_CONFIGURATION_TYPE}/lib/clang/${LLVM_VERSION}/include")
+			set (CLANG_COMPILER_HEADER_SEARCH_PATH "${LLVM_BINARY_DIR}/include/clang")
 			message (STATUS "Trying to find Clang compiler headers in '${LLVM_CONFIGURATION_TYPE}' build config in directory '${CLANG_COMPILER_HEADER_SEARCH_PATH}'.")
 			if (EXISTS ${CLANG_COMPILER_HEADER_SEARCH_PATH})
 				message (STATUS "Found headers for '${LLVM_CONFIGURATION_TYPE}' build config.")
-- 
2.27.0

