--- cromfs-1.5.4/Makefile
+++ cromfs-1.5.4/Makefile
@@ -153,7 +153,6 @@
 DOCS  = doc/FORMAT README.html doc/ChangeLog doc/*.txt
 
 all-strip: all FORCE
-	- strip cromfs-driver util/mkcromfs util/unmkcromfs util/cvcromfs
 	@echo
 	@echo Finished compiling. These were created:
 	@ls -al $(PROGS)
@@ -165,9 +164,6 @@
 
 cromfs-driver-static: $(OBJS)
 	$(CXX) -static $(CXXFLAGS) -o $@ $(OBJS) $(LDFLAGS) $(LDLIBS)
-	- strip -R.comment $@
-	# Note: It does not matter if upx cannot run.
-	- upx --best $@
 
 util/mkcromfs: FORCE
 	make -C util mkcromfs
