--- cromfs.orig/Makefile
+++ cromfs/Makefile
@@ -125,7 +125,6 @@
 DOCS  = doc/FORMAT README.html doc/ChangeLog doc/*.txt
 
 all-strip: all FORCE
-	- strip cromfs-driver util/mkcromfs util/unmkcromfs util/cvcromfs
 	@echo
 	@echo Finished compiling. These were created:
 	@ls -al $(PROGS)
@@ -137,9 +136,6 @@
 
 cromfs-driver-static: $(OBJS)
 	$(CXX) -static $(CXXFLAGS) -o $@ $(OBJS) $(LDLIBS) -lpthread -lrt
-	- strip -R.comment $@
-	# Note: It does not matter if upx cannot run.
-	- upx --best $@
 
 util/mkcromfs: FORCE
 	make -C util mkcromfs
