diff -aurN a/src/eventcomm.c b/src/eventcomm.c
--- a/src/eventcomm.c	2014-05-13 09:17:46.000000000 +0800
+++ b/src/eventcomm.c	2014-06-23 00:02:49.566942336 +0800
@@ -642,6 +642,10 @@
         fingers = 2;
     else if (comm->threeFingers)
         fingers = 3;
+    else if (comm->fourFingers)
+        fingers = 4;
+    else if (comm->fiveFingers)
+        fingers = 5;
 
     if (priv->has_touch && proto_data->num_touches > fingers)
         fingers = proto_data->num_touches;
@@ -742,6 +746,12 @@
             case BTN_TOOL_TRIPLETAP:
                 comm->threeFingers = v;
                 break;
+	    case BTN_TOOL_QUADTAP:
+		comm->fourFingers = v;
+		break;
+	    case BTN_TOOL_QUINTTAP:
+		comm->fiveFingers = v;
+		break;
             case BTN_TOUCH:
                 if (!priv->has_pressure)
                     hw->z = v ? para->finger_high + 1 : 0;
diff -aurN a/src/synproto.h b/src/synproto.h
--- a/src/synproto.h	2014-05-13 09:17:46.000000000 +0800
+++ b/src/synproto.h	2014-06-23 00:03:17.165941119 +0800
@@ -93,6 +93,8 @@
     Bool oneFinger;
     Bool twoFingers;
     Bool threeFingers;
+    Bool fourFingers;
+    Bool fiveFingers;
 };
 
 struct _SynapticsParameters;
