--- Makefile	2009-08-17 18:05:58.834309704 +0800
+++ Makefile	2009-08-17 18:06:09.574061522 +0800
@@ -84,7 +84,7 @@
 AWK = gawk
 CC = x86_64-pc-linux-gnu-gcc
 CCDEPMODE = depmode=gcc3
-CFLAGS = -march=core2 -mtune=native -O2 -pipe -mmmx -mfpmath=sse -msse2 -mssse3 -pedantic -fstrength-reduce -fexpensive-optimizations -finline-functions -funroll-loops -foptimize-register-move -funroll-loops  -fpeel-loops -ftree-loop-linear -funswitch-loops -ftree-vectorize -minline-all-stringops -fivopts --fast-math -lm -Wall -Werror -std=c99
+CFLAGS = -march=core2 -mtune=native -O2 -pipe -mmmx -mfpmath=sse -msse2 -mssse3 -pedantic -fstrength-reduce -fexpensive-optimizations -finline-functions -funroll-loops -foptimize-register-move -funroll-loops  -fpeel-loops -ftree-loop-linear -funswitch-loops -ftree-vectorize -minline-all-stringops -fivopts --fast-math -lm -Wall -std=c99
 CPP = x86_64-pc-linux-gnu-gcc -E
 CPPFLAGS = 
 CYGPATH_W = echo
