Index: opencascade-6.7.1/inc/OpenGl_GlCore11.hxx
===================================================================
--- opencascade-6.7.1.orig/inc/OpenGl_GlCore11.hxx
+++ opencascade-6.7.1/inc/OpenGl_GlCore11.hxx
@@ -34,6 +34,7 @@
 
 // exclude modern definitions and system-provided glext.h, should be defined before gl.h inclusion
 #define GL_GLEXT_LEGACY
+#define GLX_GLEXT_LEGACY
 
 // include main OpenGL header provided with system
 #if defined(__APPLE__)
Index: opencascade-6.7.1/src/OpenGl/OpenGl_GlCore11.hxx
===================================================================
--- opencascade-6.7.1.orig/src/OpenGl/OpenGl_GlCore11.hxx
+++ opencascade-6.7.1/src/OpenGl/OpenGl_GlCore11.hxx
@@ -34,6 +34,7 @@
 
 // exclude modern definitions and system-provided glext.h, should be defined before gl.h inclusion
 #define GL_GLEXT_LEGACY
+#define GLX_GLEXT_LEGACY
 
 // include main OpenGL header provided with system
 #if defined(__APPLE__)
