--- a/setup.py	2019-01-17 15:40:20.000000000 -0800
+++ a/setup.py	2019-01-23 09:40:21.136115824 -0800
@@ -3,17 +3,8 @@
 import sys
 from setuptools import setup, find_packages
 
-GIT_DESCRIBE_CMD = 'git describe --dirty --tags --long --match tinyprog-*.*'
-
 full_version_path = os.path.join(
     os.path.dirname(__file__), "tinyprog", "full_version.py")
-try:
-    full_version = subprocess.check_output(GIT_DESCRIBE_CMD, shell=True)
-    with open(full_version_path, "w") as fh:
-        fh.write('__full_version__ = "%s"\n' % full_version.strip())
-except subprocess.CalledProcessError as e:
-    sys.stderr.write(str(e))
-    sys.stderr.write('\n')
 
 with open("README.md", "rb") as fh:
     long_description = fh.read().decode('utf-8')
@@ -35,7 +26,6 @@
     use_scm_version={
         "root": "..",
         "relative_to": __file__,
-        'git_describe_command': GIT_DESCRIBE_CMD,
     },
     setup_requires=['setuptools_scm'],
     description=short_description,
