diff --git a/Makefile b/Makefile
index 03373bb..9930257 100644
--- a/Makefile
+++ b/Makefile
@@ -1,8 +1,8 @@
 VERSION = 2
 PATCHLEVEL = 6
 SUBLEVEL = 30
-EXTRAVERSION =
-NAME = Man-Eating Seals of Antiquity
+EXTRAVERSION = -coldnew
+NAME = What's this?

 # *DOCUMENTATION*
 # To see a list of typical targets execute "make help"
@@ -333,13 +333,12 @@ CHECK             = sparse
 CHECKFLAGS     := -D__linux__ -Dlinux -D__STDC__ -Dunix -D__unix__ \
                  -Wbitwise -Wno-return-void $(CF)
 MODFLAGS       = -DMODULE
-CFLAGS_MODULE   = $(MODFLAGS)
-AFLAGS_MODULE   = $(MODFLAGS)
-LDFLAGS_MODULE  =
-CFLAGS_KERNEL  =
+CFLAGS_MODULE  = $(MODFLAGS) $(CUSTOM_CFLAGS)
+AFLAGS_MODULE  = $(MODFLAGS)
+LDFLAGS_MODULE =
+CFLAGS_KERNEL  = $(CUSTOM_CFLAGS)
 AFLAGS_KERNEL  =

-
 # Use LINUXINCLUDE when you must reference the include/ directory.
 # Needed to be compatible with the O= option
 LINUXINCLUDE    := -Iinclude \
@@ -349,6 +348,9 @@ LINUXINCLUDE    := -Iinclude \

 KBUILD_CPPFLAGS := -D__KERNEL__

+# Apply custom flags
+KBUILD_CFLAGS    += $(CUSTOM_CFLAGS)
+
 KBUILD_CFLAGS   := -Wall -Wundef -Wstrict-prototypes -Wno-trigraphs \
                   -fno-strict-aliasing -fno-common \
                   -Werror-implicit-function-declaration
@@ -362,6 +364,7 @@ export VERSION PATCHLEVEL SUBLEVEL KERNELRELEASE KERNELVERSION
 export ARCH SRCARCH CONFIG_SHELL HOSTCC HOSTCFLAGS CROSS_COMPILE AS LD CC
 export CPP AR NM STRIP OBJCOPY OBJDUMP MAKE AWK GENKSYMS PERL UTS_MACHINE
 export HOSTCXX HOSTCXXFLAGS LDFLAGS_MODULE CHECK CHECKFLAGS
+export NAME

 export KBUILD_CPPFLAGS NOSTDINC_FLAGS LINUXINCLUDE OBJCOPYFLAGS LDFLAGS
 export KBUILD_CFLAGS CFLAGS_KERNEL CFLAGS_MODULE
